ArduCopter/mode_auto.cpp

```diff
@@ -166,7 +166,8 @@ bool ModeAuto::jump_to_landing_sequence_auto_RTL(ModeReason reason)
 {
     if (mission.jump_to_landing_sequence()) {
         mission.set_force_resume(true);
-        if (set_mode(Mode::Number::AUTO, reason)) {
+        // if not already in auto switch to auto
+        if ((copter.flightmode == &copter.mode_auto) || set_mode(Mode::Number::AUTO, reason)) {
```

Comment:
You can use copter.flightmode->mode_number() to get the current flight number.
I think it is better to use the value.

Reply:
I cant because it changes with `auto_RTL` state. 

https://github.com/ArduPilot/ardupilot/blob/ca6d214bb3267562de664766a4f113ad7635e2d2/ArduCopter/mode.h#L371

Or rather I could but I would have to check for both numbers.